src/main/java/in/hp/java/userservice/controller/UserController.java

```diff
@@ -0,0 +1,46 @@
+package in.hp.java.userservice.controller;
+
+import in.hp.java.userservice.model.User;
+import in.hp.java.userservice.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @PostMapping
+    public ResponseEntity<Object> addUser(@RequestBody User user) {
+        userService.addUser(user);
+        return ResponseEntity.status(HttpStatus.CREATED).build();
+    }
+
+    @GetMapping
+    public ResponseEntity<List<User>> getUsers() {
+        return ResponseEntity.ok(userService.getUsers());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Object> getUser(@PathVariable Long id) {
+        return ResponseEntity.ok(userService.getUser(id));
+    }
+
+    @PutMapping
+    public ResponseEntity<Object> updateUser(@RequestBody User user) {
+        userService.updateUser(user);
+        return ResponseEntity.ok().build();
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Object> deleteUser(@PathVariable Long id) {
+        userService.deleteUser(id);
```

Comment:
By this we are not able to send any customization message from Response Entity. This object should be generic so we can send different messages from different methods.

Reply:
Added generic response object